price/src/main.rs

```diff
@@ -0,0 +1,156 @@
+
+#[derive(Debug, clap::Parser)]
+#[clap(version = env!("CARGO_PKG_VERSION"))]
+#[clap(about = "Helium Price Oracle Server")]
+pub struct Cli {
+    /// Optional configuration file to use. If present the toml file at the
+    /// given path will be loaded. Env variables can override the
+    /// settings in the given file.
+    #[clap(short = 'c')]
+    config: Option<path::PathBuf>,
+
+    #[clap(subcommand)]
+    cmd: Cmd,
+}
+
+impl Cli {
+    pub async fn run(self) -> Result<()> {
+        let settings = Settings::new(self.config)?;
+        self.cmd.run(settings).await
+    }
+}
+
+#[derive(Debug, clap::Subcommand)]
+pub enum Cmd {
+    Server(Server),
+}
+
+impl Cmd {
+    pub async fn run(&self, settings: Settings) -> Result<()> {
+        match self {
+            Self::Server(cmd) => cmd.run(&settings).await,
+        }
+    }
+}
+
+#[derive(Debug, clap::Args)]
+pub struct Server {}
+
+impl Server {
+    pub async fn run(&self, settings: &Settings) -> Result<()> {
+        tracing_subscriber::registry()
+            .with(tracing_subscriber::EnvFilter::new(&settings.log))
+            .with(tracing_subscriber::fmt::layer())
+            .init();
+
+        // Install the prometheus metrics exporter
+        poc_metrics::start_metrics(&settings.metrics)?;
+
+        // configure shutdown trigger
+        let (shutdown_trigger, shutdown) = triggered::trigger();
+        tokio::spawn(async move {
+            let _ = signal::ctrl_c().await;
+            shutdown_trigger.trigger()
+        });
+
+        // Initialize uploader
+        let (file_upload_tx, file_upload_rx) = file_upload::message_channel();
+        let file_upload =
+            file_upload::FileUpload::from_settings(&settings.output, file_upload_rx).await?;
+
+        let store_base_path = path::Path::new(&settings.cache);
+
+        // price generators
+        let mut hnt_price_generator =
+            PriceGenerator::new(settings, BlockchainTokenTypeV1::Hnt).await?;
+        let mut mobile_price_generator =
+            PriceGenerator::new(settings, BlockchainTokenTypeV1::Mobile).await?;
+        let mut iot_price_generator =
+            PriceGenerator::new(settings, BlockchainTokenTypeV1::Iot).await?;
+        let mut hst_price_generator =
+            PriceGenerator::new(settings, BlockchainTokenTypeV1::Hst).await?;
+
+        // price watchers
+        let hnt_watch = hnt_price_generator.receiver();
+        let mobile_watch = mobile_price_generator.receiver();
+        let iot_watch = iot_price_generator.receiver();
+        let hst_watch = hst_price_generator.receiver();
+
+        // price service
+        let hnt_svc = PriceService::new(hnt_watch)?;
+        let mobile_svc = PriceService::new(mobile_watch)?;
+        let iot_svc = PriceService::new(iot_watch)?;
+        let hst_svc = PriceService::new(hst_watch)?;
+
+        // price servers
+        let hnt_server = PriceServer::new(hnt_svc);
```

Comment:
Remove all servers and tonic dependencies.. The verifiers will get the price data from the bucket, remember?

Reply:
Removed in [a454294](https://github.com/helium/oracles/pull/400/commits/a454294be181d1c7e47fa93265e510712be6cdc0)